test/gpu/native/studies/shoc/shoc-sort.skipif

```diff
@@ -1 +1,35 @@
-CHPL_GPU==cpu
+#!/bin/bash
+# Suppress ROCm6 shoc-sort failures: https://github.com/chapel-lang/chapel/issues/26669
+check_conditions() {
+    chpl_gpu=$(printenv CHPL_GPU)
+    chpl_test_perf_label=$(printenv CHPL_TEST_PERF_LABEL)
+
+    # Skip test in CPU-as-device mode
+    if [[ "$chpl_gpu" == "cpu" ]]; then
+        return 1
+    fi
```

Comment:
These lines are new here so we keep the previous behavior of skipping in cpu-as-device mode. The other aspects of the script are the same as they were in the suppressif.